synapse/api/errors.py

```diff
@@ -50,12 +51,15 @@ class Codes(object):
 
 
 class CodeMessageException(RuntimeError):
-    """An exception with integer code and message string attributes."""
+    """An exception with integer code and message string attributes.
 
-    def __init__(self, code, msg):
-        super(CodeMessageException, self).__init__("%d: %s" % (code, msg))
+    Attributes:
+        code (int): HTTP error code
+        response_code_message (str): HTTP reason phrase. None for the default.
+    """
+    def __init__(self, code):
+        super(CodeMessageException, self).__init__("%d" % code)
```

Comment:
Because I wanted to be able to turn an HTTPResponseException into a SynapseError without having my head explode.

Previously,  for an HTTPResponseException, `msg` was being used for the HTTP reason message, and `response_code_message` was unused. For a SynapseError, `msg` was used for the matrix error message, and `response_code_message` was used for the  HTTP reason message. 

![image](https://cloud.githubusercontent.com/assets/1389908/23866924/35c2b7d4-0812-11e7-8630-d2121912fb25.png)

I think. The comments were... unclear.